frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx

```diff
@@ -410,18 +423,27 @@ export function ObjectDetailView({
       </ObjectDetailContainer>
 
       <Modal
-        isOpen={typeof executeActionId === "number"}
+        isOpen={isActionExecuteModalOpen}
         onClose={handleExecuteModalClose}
       >
         <ActionExecuteModal
-          actionId={executeActionId}
+          actionId={actionId}
           initialValues={initialValues}
           fetchInitialValues={fetchInitialValues}
           shouldPrefetch
           onClose={handleExecuteModalClose}
           onSuccess={handleActionSuccess}
         />
       </Modal>
+
+      <Modal isOpen={isDeleteModalOpen} onClose={handleDeleteModalClose}>
```

Comment:
can't you move `Modal` wrapper into `DeleteObjectModal`?

Reply:
I wanted to be consistent with the current code:
- the `ActionExecuteModal` in lines above it
- there are lots (dozens) of components called `XYZModal` that render `<ModalContent>` as their root DOM element (examples: `CollectionMoveModal`, `AddToDashSelectDashModal`, `QuestionSavedModal`)

Reply:
it's not straightforward, but I agree with your points